db/migrate/20230630162646_add_user_attributes.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Add more attributes to User which can be fetched from authenticators
+class AddUserAttributes < ActiveRecord::Migration[7.0]
+  def change
+    change_table :users, bulk: true do |t|
+      t.string :provider_username
```

Comment:
**nitpick (blocking):** I'm not sure what the utility of having `provider_username` here is. Do you know what values azure and saml will provide? If we do want to add a `username` field I would assume that all users would have it.

Reply:
So with saml/azure, the username is what the user uses to sign in, and is not connected to email. For example when we sign in with azure our username is the internal science network username@domain, and the email field is firstname.lastname@phac-aspc.

Having this username value might be important for knowing that relation between user/email/internal-username, but functionally I don't think it is needed for any IRIDA functions. Maybe this would fall under the same category as phone_number where we don't bother with keeping this information with the user?

Reply:
Is that the value that we are populating the `uid` column with?

Reply:
For SAML uid and username are the same, but for Azure username is the science network username@domain but uid is a long hex value

Reply:
In that case I think we can remove the `provider_usesrname` column as well.